data-prepper-core/src/main/java/com/amazon/dataprepper/peerforwarder/PeerForwarderConfiguration.java

```diff
@@ -0,0 +1,181 @@
+        this.sslKeyFile = sslKeyFile != null ? sslKeyFile : "";
+        setBatchSize(batchSize);
+        setBufferSize(bufferSize);
+    }
+
+    @AssertTrue(message = "ssl_certificate_file must be a valid file path when ssl is enabled")
+    boolean isSslCertificateFileValid() {
+        return !ssl || isValidFilePath(sslCertificateFile);
+    }
+
+    @AssertTrue(message = "ssl_key_file must be a valid file path when ssl is enabled")
+    boolean isSslKeyFileValid() {
+        return !ssl || isValidFilePath(sslKeyFile);
+    }
+
+    private static boolean isValidFilePath(final String filePath) {
+        return filePath != null && !filePath.isEmpty() && Files.exists(Paths.get(filePath));
+    }
+
+    public int getServerPort() {
+        return serverPort;
+    }
+
+    public int getRequestTimeout() {
+        return requestTimeout;
+    }
+
+    public int getThreadCount() {
+        return threadCount;
+    }
+
+    public int getMaxConnectionCount() {
+        return maxConnectionCount;
+    }
+
+    public int getMaxPendingRequests() {
+        return maxPendingRequests;
+    }
+
+    public boolean isSsl() {
+        return ssl;
+    }
+
+    public String getSslCertificateFile() {
+        return sslCertificateFile;
+    }
+
+    public String getSslKeyFile() {
+        return sslKeyFile;
+    }
+
+    public DiscoveryMode getDiscoveryMode() {
+        return discoveryMode;
+    }
+
+    public int getBatchSize() {
+        return batchSize;
+    }
+
+    public int getBufferSize() {
+        return bufferSize;
+    }
+
+    private void setServerPort(final Integer serverPort) {
+        if (serverPort <= 0) {
+            throw new IllegalArgumentException("Server port must be a positive integer");
+        }
+        this.serverPort = serverPort;
+    }
+
+    private void setRequestTimeout(final Integer requestTimeout) {
+        if (serverPort <= 0) {
+            throw new IllegalArgumentException("Request timeout must be a positive integer");
+        }
+        this.requestTimeout = requestTimeout;
+    }
+
+    private void setThreadCount(final Integer threadCount) {
+        if (serverPort <= 0) {
+            throw new IllegalArgumentException("Thread count must be a positive integer");
+        }
+        this.threadCount = threadCount;
+    }
+
+    private void setMaxConnectionCount(final Integer maxConnectionCount) {
+        if (serverPort <= 0) {
+            throw new IllegalArgumentException("Maximum connection count must be a positive integer");
+        }
+        this.maxConnectionCount = maxConnectionCount;
+    }
+
+    private void setMaxPendingRequests(final Integer maxPendingRequests) {
+        if (serverPort <= 0) {
+            throw new IllegalArgumentException("Maximum pending requests must be a positive integer");
+        }
+        this.maxPendingRequests = maxPendingRequests;
+    }
+
+    private void setSsl(final Boolean ssl) {
+        if (ssl != null) {
```

Comment:
We should notify the invalidity to user about null value instead of swallowing it here.

Reply:
Why do we have to notify if user doesn't define the value in configuration? We will log it when we actually use SSL if it's enabled or not.

Reply:
Oh. I see you are using set methods in constructor. So null value means default.